source/docs/software/troubleshooting/robots-shouldnt-quit.rst

```diff
@@ -0,0 +1,415 @@
+
+For example, consider the following code:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 19
+
+            PWMSparkMax armMotorCtrl;
+
+            @Override
+            public void robotInit() {
+
+                armMotorCtrl.setInverted(true);
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+
+         //TODO
+
+When run, you'll see output that looks like this:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: text
+
+        ********** Robot program starting **********
+        Error at frc.robot.Robot.robotInit(Robot.java:24): Unhandled exception: java.lang.NullPointerException
+                at frc.robot.Robot.robotInit(Robot.java:24)
+                at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:94)
+                at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:335)
+                at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:387)
+                at java.base/java.lang.Thread.run(Thread.java:834)
+
+        Warning at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:350): Robots should not quit, but yours did!
+        Error at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:352): The startCompetition() method (or methods called by it) should have handled the exception above.
+
+   .. group-tab:: C++
+
+      .. code-block:: text
+
+          TODO
+
+Reading the stack trace, you can see that the issue happened inside of the ``robotInit()`` function, on line 24, and the exception involved "Null Pointer". 
+
+By going to line 24, you can see there is only one thing which could be null - ``armMotorCtrl``. Looking further up, you can see that the ``armMotorCtrl`` object is declared, but never instantiated.
+
+Alternatively, you can step through lines of code with the single step debugger, and stop when you hit line 24. Inspecting the ``armMotorCtrl`` object at that point would show that it is null. 
+
+Fixing Null Object Issues
+"""""""""""""""""""""""""
+
+Generally, you will want to ensure each reference has been initialized before using it. In this case, there is a missing line of code to instantiate the ``armMotorCtrl`` before calling the ``setInverted()`` method. 
+
+A functional implementation could look like this: 
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 19
+
+            PWMSparkMax armMotorCtrl;
+
+            @Override
+            public void robotInit() {
+                
+                armMotorCtrl = new PWMSparkMax(0);
+                armMotorCtrl.setInverted(true);
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+
+         //TODO
+
+
+Divide by Zero
+^^^^^^^^^^^^^^
+
+It is not generally possible to divide an integer by zero, and expect reasonable results. Most processors (including the RoboRIO) will cause an Unhandled Exception.

```

Comment:
As far as wording goes, I say the processor itself isn't causing the exception.
```suggestion
It is not generally possible to divide an integer by zero, and expect reasonable results. Most processors (including the RoboRIO) will raise an Unhandled Exception.
```

Reply:
Agreed, "Raise" is better here. 

I'll do some thinking as well as we start to reword for C++ too, as the core concept is straighforward, but nuanced:

The _problem_ is some logical inconsistency in the user's code, which puts the processor (or VM) in a state where it doesn't make sense to continue executing.
The exception users observe is a _result_ of the processor, the VM, or some code going into "we have a problem" mode, stopping the normal flow of execution, and raising the exception up.